File: benchmarks/pandas/bench_str_lower_upper.py
"""Benchmark: str_lower_upper — str.lower and str.upper on 100k strings"""
import json, time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = [f"Hello_World_{i % 200}" for i in range(ROWS)]
s = pd.Series(data)

for _ in range(WARMUP):
    s.str.lower()
    s.str.upper()

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.str.lower()
    s.str.upper()
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "str_lower_upper",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_str_lower_upper.ts
/**
 * Benchmark: str_lower_upper — str.lower and str.upper on 100k strings
 */
import { Series } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data = Array.from({ length: ROWS }, (_, i) => `Hello_World_${i % 200}`);
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) {
  s.str.lower();
  s.str.upper();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  s.str.lower();
  s.str.upper();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "str_lower_upper",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
